sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -2098,7 +2098,22 @@ void CAFMaker::produce(art::Event& evt) noexcept {
       }
 
       const larpandoraobj::PFParticleMetadata *pfpMeta = (fmPFPMeta.at(iPart).empty()) ? NULL : fmPFPMeta.at(iPart).at(0).get();
-      FillPFPVars(thisParticle, primary, pfpMeta, thisPFPT0, pfp);
+      const caf::PFOCharLabelsStruct pfoCharLabels {
+        fParams.PFOCharLabels().EndFractionName(),
+        fParams.PFOCharLabels().FractionalSpreadName(),
+        fParams.PFOCharLabels().DiffStraightLineMeanName(),
+        fParams.PFOCharLabels().LengthName(),
+        fParams.PFOCharLabels().MaxFitGapLengthName(),
+        fParams.PFOCharLabels().SlidingLinearFitRMSName(),
+        fParams.PFOCharLabels().AngleDiffName(),
+        fParams.PFOCharLabels().SecondaryPCARatioName(),
+        fParams.PFOCharLabels().TertiaryPCARatioName(),
+        fParams.PFOCharLabels().VertexDistanceName(),
+        fParams.PFOCharLabels().HaloTotalRatioName(),
+        fParams.PFOCharLabels().ConcentrationName(),
+        fParams.PFOCharLabels().ConicalnessName()
+      };
```

Comment:
If using an `OptionalTable`, you would need to explicitly provide the default value if no table is specified at all. For example:
```suggestion
      PFOCharLabels_t const& chargeParams
        = fParams.PFOCharLabels().value_or(PFOCharLabels_t{});
      const caf::PFOCharLabelsStruct pfoCharLabels {
        chargeParams.EndFractionName(),
        chargeParams.FractionalSpreadName(),
        chargeParams.DiffStraightLineMeanName(),
        chargeParams.LengthName(),
        chargeParams.MaxFitGapLengthName(),
        chargeParams.SlidingLinearFitRMSName(),
        chargeParams.AngleDiffName(),
        chargeParams.SecondaryPCARatioName(),
        chargeParams.TertiaryPCARatioName(),
        chargeParams.VertexDistanceName(),
        chargeParams.HaloTotalRatioName(),
        chargeParams.ConcentrationName(),
        chargeParams.ConicalnessName()
      };
```

Reply:
This worked. Thanks a lot Gianluca!